linux-kernel: update to 6.17.7

MingcongBai opened the pull request:
Topic Description
-----------------

- kernel-tools: update to 6.17.7
    Sync patches with runtime-kernel/linux-kernel.
- linux+kernel: update to 6.17.7
- linux-kernel: update to 6.17.7
    - Use \`kmsg' DRM_PANIC type to make panic messages more useful, the QR
    code mode may be enabled later as we implement a pastebin API that
    works with a base URL + QR data.
    - Refresh all kernel configurations due to Rust upgrade.
    - Track patches at AOSC-Tracking/linux @ aosc/v6.17.7
    \(HEAD: 24e7d2d3240719ae9cd621b5e341de135f4a28df\).
- kernel-tools: update to 6.17.6-rc1
    Sync patches with runtime-kernel/linux-kernel.
- linux+kernel: update to 6.17.6
- linux-kernel: update to 6.17.6-rc1
    Track patches at AOSC-Tracking/linux @ aosc/v6.17.6-rc1
    \(HEAD: 24e7d2d3240719ae9cd621b5e341de135f4a28df\).
- kernel-tools: update to 6.17.5
    Sync patches from runtime-kernel/linux-kernel.
- linux+kernel: update to 6.17.5
- linux-kernel: update to 6.17.5
    - Refresh Loongson UART patchset to v6.
    - Refresh Surface patchset to 6.17.
    - Refresh SG2042/2044 patchset.\[^1\]
    - Backport an upstream fix for ACPI access error on RISC-V ACPI+DT
    platforms.\[^2\]
    - Refresh kernel configurations.
    - Revert to using deepin-origin KVM fixes for 3B6000 and 3C6000 families.
    - \(loongarch64\) Backport a fix for rust-bindgen failure.
    - Track patches at AOSC-Tracking/linux @ aosc/v6.17.5-
    \(HEAD: 84c080e43bd2970b617f9c5a7e44224791c9be21\).
    \[^1\]: https://github.com/AOSC-Tracking/linux/pull/13
    \[^2\]: https://github.com/AOSC-Tracking/linux/pull/14
- zfs: fix build on Linux 6.17
    - Backport patches to fix Linux 6.17 support.
    - Track patches at AOSC-Tracking/zfs @ aosc/zfs-2.3.4
    \(HEAD: c78b9c57fe13119e1d927fbd825969ef3da17d4f\).

Package(s) Affected
-------------------

- kernel-tools: 6.17.7-1
- linux+kernel: 3:6.17.7
- linux-kernel-${__VER}: 6.17.7-1
- zfs: 2.3.4-2

Security Update?
----------------

No

Build Order
-----------

```
#buildit zfs linux-kernel linux+kernel kernel-tools
```

Test Build(s) Done
------------------

**Primary Architectures**

- [x] AMD64 `amd64`
- [x] AArch64 `arm64`
- [x] LoongArch 64-bit `loongarch64`
- [ ] LoongArch 64-bit (No SIMD) `loongarch64_nosimd`

**Secondary Architectures**

- [x] Loongson 3 `loongson3`
- [x] PowerPC 64-bit (Little Endian) `ppc64el`
- [x] RISC-V 64-bit `riscv64`

MingcongBai committed:
zfs: fix build on Linux 6.17

- Backport patches to fix Linux 6.17 support.
- Track patches at AOSC-Tracking/zfs @ aosc/zfs-2.3.4
  (HEAD: c78b9c57fe13119e1d927fbd825969ef3da17d4f).

MingcongBai committed:
linux-kernel: update to 6.17.5

- Refresh Loongson UART patchset to v6.
- Refresh Surface patchset to 6.17.
- Refresh SG2042/2044 patchset.[^1]
- Backport an upstream fix for ACPI access error on RISC-V ACPI+DT
  platforms.[^2]
- Refresh kernel configurations.
- Revert to using deepin-origin KVM fixes for 3B6000 and 3C6000 families.
- (loongarch64) Backport a fix for rust-bindgen failure.
- Track patches at AOSC-Tracking/linux @ aosc/v6.17.5-
  (HEAD: 84c080e43bd2970b617f9c5a7e44224791c9be21).

[^1]: https://github.com/AOSC-Tracking/linux/pull/13
[^2]: https://github.com/AOSC-Tracking/linux/pull/14

MingcongBai committed:
linux+kernel: update to 6.17.5

MingcongBai committed:
kernel-tools: update to 6.17.5

Sync patches from runtime-kernel/linux-kernel.

MingcongBai committed:
linux-kernel: update to 6.17.6-rc1

Track patches at AOSC-Tracking/linux @ aosc/v6.17.6-rc1
(HEAD: 24e7d2d3240719ae9cd621b5e341de135f4a28df).

MingcongBai committed:
linux+kernel: update to 6.17.6

MingcongBai committed:
kernel-tools: update to 6.17.6-rc1

Sync patches with runtime-kernel/linux-kernel.

grayawa commented:
zfs tested on x86_64
<img width="1042" height="120" alt="屏幕截图_20251028_205720" src="https://github.com/user-attachments/assets/6c10c235-99b4-4621-82b7-123bfc1f2ec2" />

MingcongBai committed:
linux-kernel: update to 6.17.7

- Use `kmsg' DRM_PANIC type to make panic messages more useful, the QR
  code mode may be enabled later as we implement a pastebin API that
  works with a base URL + QR data.
- Refresh all kernel configurations due to Rust upgrade.
- Track patches at AOSC-Tracking/linux @ aosc/v6.17.7
  (HEAD: 24e7d2d3240719ae9cd621b5e341de135f4a28df).

MingcongBai committed:
linux+kernel: update to 6.17.7

MingcongBai committed:
kernel-tools: update to 6.17.7

Sync patches with runtime-kernel/linux-kernel.

HmnSn commented on app-admin/zfs/spec:
```diff
@@ -1,5 +1,5 @@
-SRCS="tbl::https://github.com/openzfs/zfs/releases/download/zfs-$VER/zfs-$VER.tar.gz"
-CHKSUMS="sha256::9ec397cf360133161a1180035f3e7d6962186ed2b3457953a28d45aa883fa495"
+REL=2
+SRCS="git::commit=tags/zfs-2.3.4::https://github.com/openzfs/zfs"
```
tags/zfs-$VER

HmnSn commented:
Test on arm64, rtl8851bu works well
<img width="1186" height="733" alt="image" src="https://github.com/user-attachments/assets/e84f9169-b25d-4ebd-a841-8bd027a66861" />

ziyao233 commented:
Tested on x86_64 (Panasonic CF-SZ6), MMC card reader, WiFI, and audio are okay.

<img width="1280" height="720" alt="image" src="https://github.com/user-attachments/assets/18f82398-4765-42ac-bc29-4482cc18f61e" />

Here's mine `Tested-by: Yao Zi <ziyao@disroot.org>`.

MingcongBai committed:
topics: add linux-kernel-6.17.y.toml

MingcongBai commented:
Fixed a trivial issue in `app-admin/zfs/spec`, no rebuild planned.

MingcongBai commented:
TUM:

<img width="1088" height="161" alt="图片" src="https://github.com/user-attachments/assets/0c4dd674-e3a8-47ef-bcad-a5ec57c55281" />

MingcongBai commented:
Tested good on various x86 platforms:

- Lenovo ThinkPad E16 Gen 3 (Arrow Lake-H)
- Lenovo ThinkPad A485 (AMD Zen1 + Vega 8)
- Lenovo ThinkBook 14 G6+ (AHP, AMD Zen4 + AMD RDNA3.5)
- Lenovo Kaitian N70Z-G1d (Zhaoxin KaiXian KX-6640MA)
- HO-Z6000G (Zhaoxin KaiXian KX-6000G/4)
- QEMU 10.1.2 (UEFI)

AArch64:

- Huawei/HiSilicon Kunpeng 920-3211K
- Huawei/HiSilicon Kunpeng 920-7260
- Neoverse N2

LoongArch:

- Loongson 3A6000
- Loongson 3C5000
- Loongson 3C6000/S
- Loongson 3C6000/D
- Loongson 3C6000/Q

xry111 commented:
Tested on Loongson 3A6000M and 3B6000M as well.

grayawa commented:
add zfs test result:
![IMG_20251109_193337](https://github.com/user-attachments/assets/dbfb7c69-84c2-4b7d-ad0c-90cba4b817dd)
